doc/design/inference.md

```diff
@@ -0,0 +1,105 @@
+# Design Doc: Inferencer
+
+In fluid, a nueral network is represented as a protobuf message [ProgramDesc](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/program.md), the python wrapper of which is `Program`.

```

Comment:
Please be aware that Fluid doesn't represent a network at all.  The protobuf message represents the program, not the network.

Reply:
There is no network in Fluid. However, `neural network` seems a phrase in deep learning. I'll think about a better expression.